src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType: symbol = Symbol("UpdateStreamType");
+const CHECK_DELAY: number = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING: string = "dcosUIVersion";
+const DISMISSED_VERSION: string = "dismissedVersion";
+const LAST_TIME_CHECK: string = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+  [key: string]: DCOSUIVersion;
```

Comment:
What is this `[key: string]` part about?

Reply:
I searched for the error and I thought it was complaining about the key.

Reply:
The problem is that you are doing `savedStates[DCOS_UI_VERSION_SETTING] =  ...`. As some mutating method could have changed `DCOS_UI_VERSION_SETTING` typescript tries to tell you that there could be every key accessed. If you would change the line to `savedStates.dcosUIVersion` there would not be an error

Reply:
Yes, but I wanted to make use of the constant.

Reply:
It makes it easier to change `"dcosUIVersion"`.